src/parser/parser_env.ml

```diff
@@ -616,12 +615,13 @@ module Peek = struct
 
   let ith_comments ~i env =
     let comments = Lex_result.comments (lookahead ~i env) in
+    match comments with 
+    | [] -> []
+    | _ -> 
```

Comment:
```suggestion
    (* perf: avoid closure allocations from List.filter for the common case of empty list *)
    match comments with 
    | [] -> []
    | _ -> 
```

We might also consider (a) using Base.List.filter here, which has the optimization we want built-in, and (b) proposing the eta-expansion in the stdlib definition. It's a small change, and clearly affected this code.

Reply:
addressed in coming commits. Note this would make flow_parser depends on base though